### PR TITLE
Remove curr_global_slot from Snapp_predicate.Protocol_state

### PR DESCRIPTION
**Pull request by psteckler**
Remove `curr_global_slot` from `Snapp_predicate.Protocol_state`, since it's an internal bookkeeping detail. The remaining field `global_slot_since_genesis` provides sufficient slot information.

In a small number of places, uses of `curr_global_slot` are replaced with `global_slot_since_genesis`.

### CODE REVIEW

File: src/lib/mina_base/transaction_logic.ml [hunk 1]
```diff
@@ -1633,8 +1633,8 @@ module Make (L : Ledger_intf) : S with type ledger := L.t = struct
                    (Control.tag p.authorization))
               ~has_proof:(Control.Tag.equal (Control.tag p.authorization) Proof)
               ~is_new:(match loc with `Existing _ -> false | `New -> true)
-              ~current_global_slot:global_state.protocol_state.curr_global_slot
-              p.data a
+              ~current_global_slot:
```
Comment (mrmr1993) [inline src/lib/mina_base/transaction_logic.ml:1636]: Can you rename this argument, for completeness?

**Review by mrmr1993 — COMMENTED** [review]
Nit to ensure that somebody doesn't 'fix' this fix, but otherwise lgtm

**Review by deepthiskumar — COMMENTED** [review]
Could you update the description here https://github.com/MinaProtocol/mina/blob/f2df0e4bc93b1015d0c9c5534161aa255d4e12f4/src/lib/mina_graphql/mina_graphql.ml#L2169  as well?

File: src/lib/mina_base/transaction_logic.ml (hunk 1 above)
Comment (psteckler) [reply src/lib/mina_base/transaction_logic.ml]: Done.

**Review by deepthiskumar — CHANGES_REQUESTED** [review]
You'll also need to update the expiry check in the transaction pool https://github.com/MinaProtocol/mina/blob/32e09e41c5514c0f25a16d035a19cbcef4b3ed65/src/lib/network_pool/indexed_pool.ml#L326

**Comment by deepthiskumar** [comment]
> You'll also need to update the expiry check in the transaction pool
> 
> https://github.com/MinaProtocol/mina/blob/32e09e41c5514c0f25a16d035a19cbcef4b3ed65/src/lib/network_pool/indexed_pool.ml#L326

Actually it already does. Got confused by the name. Could you update the name there as well, please?

**Review by deepthiskumar — APPROVED** [review]

**Review by mrmr1993 — APPROVED** [review]

**Review by MartinMinkov — APPROVED** [review]

**Comment by psteckler** [comment]
Superseded by #9511.

**Pull request closed without merging**